src/vm/vm_core.rs

```diff
@@ -3601,191 +3091,53 @@ mod tests {
             )],
         );
 
-        //Create program and execution segments
-        for _ in 0..2 {
-            vm.segments.add(&mut vm.memory, None);
-        }
         //Initialzie registers
         vm.run_context.pc = MaybeRelocatable::from((0, 3));
         vm.run_context.ap = MaybeRelocatable::from((1, 2));
         vm.run_context.fp = MaybeRelocatable::from((1, 2));
+
+        //Create program and execution segments
+        for _ in 0..2 {
+            vm.segments.add(&mut vm.memory, None);
+        }
+
```

Comment:
This block of code is not needed, its already covered by the memory macro

Reply:
I removed all instances of the segment adding where I could. However, when running this particular test (test_step_for_preset_memory_with_alloc_hint), an extra segment is needed.
```rust
assert_eq!(
            vm.memory.data[2],
            vec![Some(MaybeRelocatable::from(bigint!(1)))]
        );
```